### ISSUE TITLE
External data disk support

### ISSUE THREAD

**Issue by karataliu** [Azure/service-fabric-issues#939]
I'm trying to deploy a new package to my cluster, and see the following exception
```
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Fabric.FabricException: An error occurred during this operation.  Please check the trace logs for more details. ---> System.Runtime.InteropServices.COMException: There is not enough space on the disk. (Exception from HRESULT: 0x80070070)
   at System.Fabric.Interop.NativeImageStore.IFabricNativeImageStoreClient2.UploadContent(IntPtr remoteDestination, IntPtr localSource, UInt32 timeoutMilliseconds, FABRIC_IMAGE_STORE_COPY_FLAG copyFlag)
```

When I logging on my cluster VM, I found the log dirs is consuming up all the disk space:

Windows cluster:
SvcFab/Log
SvcFab/ReplicatorLog
8G+

Linux cluster:
sfroot/log
7G+

Both cluster has been created for only one week and with no heavy load.

The default datapath is set to local disk 'D:' or '/mnt'. The problem is the local disk space is associated with VM type and is always limited.

Can we support setting datapath to an attached data disk?

**Comment by xunlu** [comment]
What is the service fabric version of your cluster?

**Comment by alexwun** [comment]
@karataliu Can you summarize which files are the largest?

And you're also making sure to remove old unused application packages and unregister unused application types (https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-deploy-remove-applications)?

**Comment by karataliu** [comment]
@xunlu 
Windows: 6.1.472.9494
Linux 6.1.186.1

@alexwun 
Looks like it's almost the trace files located on the vm
On Windows
D:\SvcFab\Log\Traces\*.etl
D:\SvcFab\ReplicatorLog\replicatorshared (8G)

On Linux:
There are about 500+ files each with 8MB size in following location:
/mnt/sfroot/log/lttng-traces/ServiceFabric_0-{id}/ust/uid/1001/64-bit/channel0_0_*


I use a modified the template based on quick start templates to deploy the cluster:
1. The VMSS size is DS2_v2 , which has a relative small local disk.
2. I didn't enable diagnostic extension, nor passing account key to service fabric extension. 

I checked another cluster with diagnostic enabled, the mentioned trace file just does not lie on the vm.

**Comment by alexwun** [comment]
@karataliu If a significant amount of space is being consumed by *.etl files, then try decreasing the MaxDiskQuotaInMB setting and increasing the DiskFullSafetySpacInMB cluster setting:

https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-cluster-fabric-settings

*.etl files are diagnostic log files, so they can be safely deleted to mitigate disk space issues. Service Fabric does limit how much disk space is used by diagnostic log files already (controlled by the two cluster settings mentioned above).

**Comment by karataliu** [comment]
Thanks @alexwun  That link is helpful.

Just a quick question regrading the behavior diagnostic config: I did see the tracing log uploaded to the blob storage account if cluster enables diagnostic. 
Does it mean if 'diagnosticsStorageAccountConfig' is set, then the tracing log would be stored in blob storage? I didn't find much doc regarding 'diagnosticsStorageAccountConfig' though.

**Comment by alexwun** [comment]
@karataliu Yes, since you are running in Azure we configure the Service Fabric support logs for you automatically:

https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-diagnostics-event-generation-infra

The support logs are generated locally on each node and periodically uploaded to Azure storage automatically - the local logs are eventually deleted from the nodes. Typically, nothing needs to be configured for these support logs, but since you are using smaller VMs that are running low on disk space, these two settings can be used to limit amount of disk space used locally on the nodes for support logs.

**Comment by karataliu** [comment]
@alexwun Got it. Thanks for the information!
